vision/mlp_mnist/mlp_mnist.jl

```diff
@@ -100,16 +103,16 @@ function loss_and_accuracy(data_loader, model, device)
     for (x, y) in data_loader
         x, y = device(x), device(y)
         ŷ = model(x)
-        ls += logitcrossentropy(ŷ, y, agg=sum)
+        ls += loss(model, x, y)
```

Comment:
This runs the model twice, you could delete the `ŷ = model(x)` line.

Reply:
Thanks for your review, @mcabbott .

I have noticed this, but unfortunately `ŷ` is used in line 107 (new) to calculate the accuracy.

Thought it's better to have one centralized definition of loss than having `model(x)` executed twice.

Reply:
just keep the previous definition
`ls += logitcrossentropy(ŷ, y, agg=sum)`
Also notice that the aggregation here is `sum`, not the default `mean`

Reply:
Ah I see. 

Another style would be this: Just delete the definition `loss(model, x, y) = logitcrossentropy(model(x), y)` completely, and write the pieces both here and inside the gradient call. Maybe that's more explicit? 

If desired, it could have `const loss = logitcrossentropy` somewhere, so that there's one place to change it. But hard-coding it seems fine.

Reply:
Thanks again for your comments.

@CarloLucibello: I restored aggregation by `sum` again (I changed this on purpose before to have the same loss used at every place; but I did not check carefully and actually the summed loss was divided by the number of samples afterwards. Thus it also calculated the `mean` within `loss_and_accuracy` and my previous commit introduced a bug.

@mcabbott: I decided for [this commit](https://github.com/FluxML/model-zoo/pull/385/commits/dacacb9859fbbe327edafe32d2a38d96a3a5cd6b) to account for your and @CarloLucibello's suggestions. Hope that's fine now and sorry for the temporarily introduced mini-bug regarding loss output.

Reply:
Done, thanks for fixing this up!